core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentPropertiesAndSchemaHolder.java

```diff
@@ -106,14 +111,16 @@ public int addSegmentProperties(CarbonTable carbonTable,
     SegmentIdAndSegmentPropertiesIndexWrapper segmentIdSetAndIndexWrapper =
         this.segmentPropWrapperToSegmentSetMap.get(segmentPropertiesWrapper);
     if (null == segmentIdSetAndIndexWrapper) {
-      synchronized (getOrCreateTableLock(carbonTable.getAbsoluteTableIdentifier())) {
+      synchronized (getOrCreateTableLock(
+          segmentPropertiesWrapper.carbonTable.getAbsoluteTableIdentifier())) {
         segmentIdSetAndIndexWrapper =
             this.segmentPropWrapperToSegmentSetMap.get(segmentPropertiesWrapper);
         if (null == segmentIdSetAndIndexWrapper) {
           // create new segmentProperties
           segmentPropertiesWrapper.initSegmentProperties();
           segmentPropertiesWrapper.addMinMaxColumns(carbonTable);
           int segmentPropertiesIndex = segmentPropertiesIndexCounter.incrementAndGet();
+          updateDeleteMap(segmentIdSetAndIndexWrapper.getSegmentPropertiesIndex(), -1);
```

Comment:
Is it needed to add -1 while adding? I feel it is better  we can add only when invalidate request comes